go/test/endtoend/reparent/semisync/semi_sync_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semisync
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/test/endtoend/reparent/utils"
+)
+
+func TestSemiSyncUpgradeDowngrade(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	clusterInstance := utils.SetupReparentCluster(t, "semi_sync")
+	defer utils.TeardownCluster(clusterInstance)
+	tablets := clusterInstance.Keyspaces[0].Shards[0].Vttablets
+
+	// Verify that replication is running as intended.
+	utils.ConfirmReplication(t, tablets[0], []*cluster.Vttablet{tablets[1], tablets[2], tablets[3]})
+
+	replica := tablets[1]
+	// Verify we are using the correct vttablet version.
+	verifyVttabletVersion(t, replica, 21)
```

Comment:
it feels like this will fail once we upgrade `main` to `v22.0.0-SNAPSHOT`

Reply:
At that point, I think we'll just remove the test because everything will be using new semi-sync plugins and I don't see the need to keep this test anymore.

Reply:
If you would prefer, I can add a skip line that doesn't run the test if the version isn't v21 exactly. In this case, it wouldn't fail, but would be skipped when we go to v22.0-snapshot.

Reply:
> If you would prefer, I can add a skip line that doesn't run the test if the version isn't v21 exactly. In this case, it wouldn't fail, but would be skipped when we go to v22.0-snapshot.

If we plan on removing it in `v22` it makes sense to do this. That way CI will not fail by surprise on `main` in the future.

Reply:
Yep, I'll make that change 👍